src/realm/util/cli_args.hpp

```diff
@@ -0,0 +1,104 @@
+#ifndef REALM_UTIL_CLI_ARGS_HPP
+#define REALM_UTIL_CLI_ARGS_HPP
+
+#include <vector>
+
+#include "realm/util/string_view.hpp"
+
+namespace realm::util {
+
+class CliFlag;
+class CliArgumentParser {
```

Comment:
Any reason not to use `getopt_long()` as we do for other CLI utils, other than the fact that it is a trainwreck?